traffic_portal/app/src/common/modules/table/profilesParamsCompare/TableProfilesParamsCompareController.js

```diff
@@ -0,0 +1,213 @@
+			controller: 'DialogConfirmController',
+			size: 'md',
+			resolve: {
+				params: function () {
+					return params;
+				}
+			}
+		});
+		modalInstance.result
+			.then(
+				function() { updateProfile1 = true; },
+				function() { updateProfile1 = false; }
+			)
+			.finally(
+				function() {
+					let params = {
+						title: 'Modify ' + profile2.name + ' parameters',
+						message: 'The ' + profile2.name + ' profile is used by ' + this.profile2Usage + '. Are you sure you want to modify the parameters?'
+					};
+					let modalInstance = $uibModal.open({
+						templateUrl: 'common/modules/dialog/confirm/dialog.confirm.tpl.html',
+						controller: 'DialogConfirmController',
+						size: 'md',
+						resolve: {
+							params: function () {
+								return params;
+							}
+						}
+					});
+					modalInstance.result
+						.then(
+							function() { updateProfile2 = true; },
+							function() { updateProfile2 = false; }
+						)
+						.finally(
+							function() {
+								let promises = [];
+
+								if (updateProfile1) {
+									promises.push(profileParameterService.linkProfileParameters(profile1, _.pluck($scope.selectedProfile1Params, 'id')));
+								}
+
+								if (updateProfile2) {
+									promises.push(profileParameterService.linkProfileParameters(profile2, _.pluck($scope.selectedProfile2Params, 'id')));
+								}
+
+								if (promises.length > 0) {
+									$q.all(promises)
+										.then(
+											function(result) {
+												let messages = [];
+												for (let i = 0; i < result.length; i++) {
+													messages = _.union(messages, result[i].data.alerts);
+												};
+												messageModel.setMessages(messages, false);
+											})
+										.finally(
+											function() {
+												$scope.refresh();
+											}
+										);
+								}
+							}
+						);
+				}
+			);
+	};
+
+	$scope.dirty = false;
+
+	$scope.profile1 = profile1;
+	$scope.profile2 = profile2;
+	$scope.profilesParams = profilesParams;
+
+	$scope.selectedProfile1Params = [];
+	$scope.selectedProfile2Params = [];
+
+	$scope.selectedParams = _.map(profilesParams, function(param) {
+		let isAssignedToProfile1 = _.find(profile1.params, function(profile1param) { return profile1param.id == param.id }),
+			isAssignedToProfile2 = _.find(profile2.params, function(profile2param) { return profile2param.id == param.id });
+
+		if (isAssignedToProfile1) {
+			param['origSelected1'] = true;
+			param['selected1'] = true;
+		} else {
+			param['origSelected1'] = false;
+			param['selected1'] = false;
+		}
+
+		if (isAssignedToProfile2) {
+			param['origSelected2'] = true;
+			param['selected2'] = true;
+		} else {
+			param['origSelected2'] = false;
+			param['selected2'] = false;
+		}
+
+		return param;
+	});
+
+	$scope.updateSelectedCount = function(profNum) {
+		$scope['selectedProfile' + profNum + 'Params'] = _.filter($scope.selectedParams, function(param) { return param['selected' + profNum] == true; } );
```

Comment:
Instead of using `_.filter`, consider using [the built-in `Array.prototype.filter` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) like so:
```javascript
// assuming 'profNum' is a Number, it should be parsed to avoid type coercion mishaps
const num = parseInt(profNum, 10);
$scope['selectedProfile'+num+'Params'] = $scope.selectedParams.filter(function(p){return param['selected'+num];});
```